### PR TITLE
FI-1650: Remove Warning Message from Limited App Launch

### PR DESCRIPTION
**Pull request by Jack-Fraser16**
Addresses [this issue](https://github.com/onc-healthit/onc-certification-g10-test-kit/issues/155) by using config to skip the `check_for_missing_scopes` warning in the Limited Access App group. 

~~Less than ideal solution because it addresses a G10-specific issue within the SMART repo. But, short of replicating this test in G10 within the Limited Access group (which I guess is an option?), this is the best I could come up with and it's a minor change.~~

I tested this change on G10. Here's **2.1.06** passing without warnings:

Screen Shot 2022-07-27 at 10 40 58 AM

And here's the warning still showing in **1.2.06** when all the requested scopes are not given:

Screen Shot 2022-07-27 at 10 37 23 AM

### CODE REVIEW

File: lib/smart_app_launch/token_response_body_test.rb
Comment (Jammjammjamm) [inline lib/smart_app_launch/token_response_body_test.rb:46]: Add a `skip_missing_scopes_check` config option and use that to skip this.

Comment (arscan) [reply lib/smart_app_launch/token_response_body_test.rb]: You should  update the README to reflect this new config option as you are essentially updating this test's "API": https://github.com/inferno-framework/smart-app-launch-test-kit#standalone-launch-group

I do wonder if it makes sense for the default here to expect scopes requested equal scopes received, and to provide a warning if not (implying that servers SHOULD be granting scopes equal to those requested).  Implying that servers are doing something wrong by allowing users to deny scopes is not correct, unless there are additional non-SMART expectations being added here.  Which I would argue are not appropriate for these base tests.

Seems like the default should be to provide no expectations.  And then allow the test writers to layer on additional expectations in their own contexts.  Either through a config (fail_if_any_scopes_denied; warn_if_any_scopes_denied), or just having the pattern be that you add another test that checks the scopes_requested / scopes_received inputs/outputs in your own context (which is what we are doing in g10, right?  these warnings don't actually drive anything meaningful)?

I'll defer to Steve as he's architected these, but my vote would be to just downgrade the warnings to infos and not have these tests be concerned with context-specific expectations.

File: lib/smart_app_launch/token_response_body_test.rb
Comment (Jammjammjamm) [inline lib/smart_app_launch/token_response_body_test.rb:46]: The meaning of `limited launch` is far from obvious. You should give this configuration option a name that says what it does.

Comment (arscan) [reply lib/smart_app_launch/token_response_body_test.rb:46]: @Jammjammjamm I provided similar feedback (and more) but visibility may have been lost when the line of code changed?  I'm curious what you think here.

> You should update the README to reflect this new config option as you are essentially updating this test's "API": https://github.com/inferno-framework/smart-app-launch-test-kit#standalone-launch-group
> 
> I do wonder if it really makes sense for the default here to expect scopes requested equal scopes received, and to provide a warning if not (implying that servers SHOULD be granting scopes equal to those requested). Implying that servers are doing something wrong by allowing users to deny scopes is not correct, unless there are additional scenario/context-specific expectations being added here. Which I would argue are not appropriate for these base tests.
> 
> Seems like the default should be to provide no expectations. And then allow the test writers to layer on additional expectations in their own contexts. Either through a config (fail_if_any_scopes_denied; warn_if_any_scopes_denied), or just having the pattern be that you add another test that checks the scopes_requested / scopes_received inputs/outputs in your own context (which is what we are doing in g10, right? these warnings don't actually drive anything meaningful)?
> 
> I'll defer to Steve as he's architected these, but my vote would be to just downgrade the warnings to infos (at most) and not have these tests be concerned with context-specific expectations.

Comment (Jammjammjamm) [reply lib/smart_app_launch/token_response_body_test.rb:46]: While it's not wrong to allow users to deny scopes, it seems weird to me to deny scopes in a testing session unless you're explicitly testing the ability to deny scopes. I think a missing scope is most likely to be caused by some sort of configuration error and it makes sense to have them be very visible by default.

Comment (arscan) [reply lib/smart_app_launch/token_response_body_test.rb]: We should have this soft expectation documented in the test description so the tester knows that they should approve all scopes if prompted (depending on this config flag).  Otherwise we are inventing a should requirement because that's definitely not coming from a spec (a system should grant all scopes -- no).

Comment (Jack-Fraser16) [reply lib/smart_app_launch/token_response_body_test.rb]: @arscan added

File: README.md
Comment (Jammjammjamm) [inline README.md:12]:
```suggestion
- Navigate to `http://localhost`. The SMART test suite will be available.
```

File: lib/smart_app_launch/token_response_body_test.rb
Comment (arscan) [inline lib/smart_app_launch/token_response_body_test.rb:14]: I'm not sure we should have this language in here like this because the test user can't be expected to locate the source and inspect code to determine the behavior of the test.

If we can't change hte language in the description based on if the flag is set or not, I'd vote to leave the language out.  I know that means we endsus up in the situation where we have a undocumented requirement, which is partly why I was advocating for simply not having this warning logic be put in here at all, but its not a huge deal and we just need this done for the release.

**Review by arscan — APPROVED** [review]
I think this is good.  I think we should remove that added language that references a config option, as a tester cannot be expected to know what the test code itself looks like, but this isn't a big deal and we should just get this in.

**Pull request merged**
